lib/widgets/components/purchase/purchase_process.dart

```diff
@@ -0,0 +1,116 @@
+import 'package:coffeecard/base/strings.dart';
+import 'package:coffeecard/base/style/colors.dart';
+import 'package:coffeecard/cubits/purchase/purchase_cubit.dart';
+import 'package:coffeecard/errors/match_case_incomplete_exception.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+class PurchaseProcess extends StatefulWidget {
+  @override
+  State<StatefulWidget> createState() => _PurchaseProcessState();
+}
+
+class _PurchaseProcessState extends State<PurchaseProcess>
+    with WidgetsBindingObserver {
+  AppLifecycleState? _notification;
+
+  @override
+  void didChangeAppLifecycleState(AppLifecycleState state) {
+    setState(() {
+      _notification = state;
+    });
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    WidgetsBinding.instance!.addObserver(this);
+  }
+
+  @override
+  void dispose() {
+    WidgetsBinding.instance!.removeObserver(this);
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Center(
+      child: Column(
+        mainAxisAlignment: MainAxisAlignment.center,
+        children: [
+          BlocBuilder<PurchaseCubit, PurchaseState>(
+            builder: (context, state) {
+              final cubit = context.read<PurchaseCubit>();
+              if (_notification == AppLifecycleState.resumed) {
+                cubit.verifyPurchase();
+              }
+              if (state is PurchaseInitial) {
+                //Not related to previous check, hence a separate if statement
+                cubit.payWithMobilePay();
+                return makeDialog(Strings.purchaseTalking);
+              } else if (state is PurchaseProcessing ||
+                  state is PurchaseStarted) {
+                return makeDialog(Strings.purchaseTalking);
+              } else if (state is PurchaseVerifying) {
+                return makeDialog(Strings.purchaseCompleting);
+              } else if (state is PurchaseCompleted) {
+                return makeDialog(Strings.purchaseSuccess);
+              } else if (state is PurchasePaymentRejected) {
+                return makeDialog(
+                  Strings.purchaseRejectedOrCanceled,
+                  content: const Text(
+                    Strings.purchaseRejectedOrCanceledMessage,
+                  ),
+                );
+              } else if (state is PurchaseError) {
+                return makeDialog(
+                  Strings.purchaseError,
+                  content: Text(state.message),
+                );
+              } else {
+                //FIXME: message
+                throw MatchCaseIncompleteException(
+                  'Unmatched state for PurchaseState',
+                );
+              }
+            },
+          ),
+        ],
+      ),
+    );
+  }
+
+  StatelessWidget makeDialog(String title, {Widget? content}) {
+    return content != null
```

Comment:
What is the difference between AlertDialog and SimpleDialog? seems like there is a bit of duplication here

Reply:
AlertDialog takes an array of widgets representing the actions that a user have, e.g. the "Ok" button. Since these are actually nullable, we could likely reuse the AlertDialog for both cases, without any issue, it just seemed to better communicate their intended purpose by using the widget designed for those cases

Reply:
allright. I suggest moving the Text(title) and shape into shared variables then

Reply:
Sure, but you are then fine with it being two separate types of widget? I thought about it for a bit more, and I don't really like how I distinguish between which widget gets created, i.e. I assume we want an alert if content is provided. Any thoughts?

Reply:
Good idea, just pushed a change to do so. Take a look and confirm whether that is what you imagined. If everything looks alright with you, feel free to merge it in.